pkg/reconciler/inmemorychannel/dispatcher/controller.go

```diff
@@ -56,8 +61,32 @@ const (
 	writeTimeout  = 15 * time.Minute
 	port          = 8080
 	finalizerName = "imc-dispatcher"
+
+	readinessProbeOK       = http.StatusNoContent
+	readinessProbeError    = http.StatusInternalServerError
+	readinessProbeNotReady = http.StatusServiceUnavailable
 )
 
+// ReadinessChecker can assert the readiness of a component.
+type ReadinessChecker interface {
+	IsReady() (bool, error)
+}
+
+// DispatcherReadyChecker asserts the readiness of a dispatcher for in-memory Channels.
+type DispatcherReadyChecker struct {
+	// Allows listing observed Channels.
+	chLister v1.InMemoryChannelLister
+
+	// Allows listing/counting the handlers which have already been registered.
+	chMsgHandler multichannelfanout.MultiChannelMessageHandler
+
+	// Allows safe concurrent read/write of 'isReady'.
+	sync.Mutex
+
+	// Minor perf tweak, bypass check if we already observed readiness once.
+	isReady bool
+}
```

Comment:
nit: Just a suggestion, but maybe we could move this type + its methods out of `controller.go`? Something like `readiness.go` or any other name that sounds suitable to you.